src/msg_processor.rs

```diff
@@ -126,13 +128,18 @@ impl MsgFormat {
     }
 }
 
-pub fn random_cob_id(msg_format: &MsgFormat) -> u32 {
+pub fn random_cob_id_with_format(msg_format: &MsgFormat) -> u32 {
```

Comment:
Good change, hadn't thought of doing that but yeah those should be different functions.